File: longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<Integer>();
        for (int num : nums) {
            set.add(num);
        }

        int longestStreak = 0;
        for (int num : set) {
            if (!set.contains(num - 1)) {
                int streak = 0;
                while(set.contains(num++)) {
                    streak++;
                }
                longestStreak = Math.max(longestStreak, streak);
            }
        }
        return longestStreak;
    }
}

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums.sort()
        streak = 0
        best = 0
        for i in range(len(nums)):
            if i == 0 or nums[i] == nums[i - 1] + 1:
                streak += 1
                best = max(best, streak)
            elif nums[i] == nums[i - 1]:
                continue
            else:
                streak = 1
        return best